constants.go

```diff
@@ -823,9 +825,15 @@ const (
 	UsageWindowsDesktopOnly = "usage:windows_desktop"
 )
 
+// ErrNodeIsAmbiguous serves as an identifying error string indicating that
+// the proxy subsystem found multiple nodes matching the specified hostname.
+var ErrNodeIsAmbiguous = &trace.NotFoundError{Message: "ambiguous host could match multiple nodes"}
+
 const (
 	// NodeIsAmbiguous serves as an identifying error string indicating that
 	// the proxy subsystem found multiple nodes matching the specified hostname.
+	// TODO(tross) DELETE IN v20.0.0
+	// Deprecated: Prefer using ErrNodeIsAmbiguous
 	NodeIsAmbiguous = "err-node-is-ambiguous"
```

Comment:
nit: I might be missing something super obvious, but I'm not seeing where/how we're preserving the meaningfulness of `ErrNodeIsAmbiguous` across network boundaries/ssh.  Is there some magic error handling I'm missing that is making sure that `errors.Is(...)` is working correctly when `ErrNodeIsAmbiguous` is generated during routing?

Reply:
Ah okay, I follow now.  `gravitational/trace` has an implementation of the `Is` method that takes message string equality into account.  Depending on entire message string equality feels a lot more brittle than depending on the presence of a single unique identifier within the message.  It basically locks us into this exact error message forever.  Also, string equality doesn't seem to usually be something that `errors.Is` cares about, and I think most people touching an error that is intended to be used with `errors.Is` wouldn't think of string equality as a potential compatibility issue.  My preference would be to leave this the way it was.